spec/ddtrace/sampler_spec.rb

```diff
@@ -0,0 +1,236 @@
+require 'spec_helper'
+
+require 'ddtrace'
+require 'ddtrace/sampler'
+
+RSpec.describe Datadog::AllSampler do
+  subject(:sampler) { described_class.new }
+
+  before(:each) { Datadog::Tracer.log.level = Logger::FATAL }
+  after(:each) { Datadog::Tracer.log.level = Logger::WARN }
+
+  describe '#sample!' do
+    let(:spans) do
+      [
+        Datadog::Span.new(nil, '', trace_id: 1),
+        Datadog::Span.new(nil, '', trace_id: 2),
+        Datadog::Span.new(nil, '', trace_id: 3)
+      ]
+    end
+
+    it 'samples all spans' do
+      spans.each do |span|
+        expect(sampler.sample!(span)).to be true
+        expect(span.sampled).to be true
+      end
+    end
+  end
+end
+
+RSpec.describe Datadog::RateSampler do
+  subject(:sampler) { described_class.new(sample_rate) }
+
+  before(:each) { Datadog::Tracer.log.level = Logger::FATAL }
+  after(:each) { Datadog::Tracer.log.level = Logger::WARN }
+
+  describe '#initialize' do
+    context 'given a sample rate' do
+      context 'that is negative' do
+        let(:sample_rate) { -1.0 }
+        it { expect(sampler.sample_rate).to eq(1.0) }
+      end
+
+      context 'that is 0' do
+        let(:sample_rate) { 0.0 }
```

Comment:
It would be weird for someone to sample at a rate of 0, but I don't see why we shouldn't allow this?

Reply:
It's just the previous logic from the old tests. I agree it's technically acceptable, maybe we can change it in the future.

Reply:
:+1: for later